tests/test_requests.py

```diff
@@ -0,0 +1,87 @@
+import json
+from unittest.mock import patch
+
+import httpretty
+import pytest
+import requests
+from flask import Flask
+from flask_multipass import Multipass
+from requests.sessions import Session
+
+from flask_multipass_cern import OIDC_RETRY_COUNT, CERNIdentityProvider
+
+
+@patch('flask_multipass_cern.CERNIdentityProvider._get_api_session')
+def test_get_identity_groups_retry(mock_get_api_session):
+    mock_get_api_session.return_value = Session()
+    app = Flask('test')
+    Multipass(app)
+
+    with app.app_context():
```

Comment:
this would nicely fit in a pytest fixture

Reply:
I was thinking exactly that, extracting that into a fixture to avoid repeating it on every test.